airbyte-integrations/connectors/destination-mysql/src/main/java/io/airbyte/integrations/destination/mysql/MySQLDestination.java

```diff
@@ -112,6 +122,61 @@ public JsonNode toJdbcConfig(final JsonNode config) {
     return Jsons.jsonNode(configBuilder.build());
   }
 
+  private List<String> getAdditionalParameters(final JsonNode config) {
+    final Map<String, String> customParameters = getCustomJdbcParameters(config);
+
+    if (useSSL(config)) {
+      return convertToJdbcStrings(customParameters, List.of(DEFAULT_JDBC_PARAMETERS, SSL_JDBC_PARAMETERS));
+    } else {
+      return convertToJdbcStrings(customParameters, List.of(DEFAULT_JDBC_PARAMETERS));
+    }
+  }
+
+  private List<String> convertToJdbcStrings(final Map<String, String> customParameters, final List<Map<String, String>> defaultParametersMaps) {
+    assertCustomParametersDontOverwriteDefaultParameters(customParameters, defaultParametersMaps);
+    return Streams.concat(Stream.of(customParameters), defaultParametersMaps.stream())
+        .map(Map::entrySet)
+        .flatMap(Collection::stream)
+        .map(entry -> formatParameter(entry.getKey(), entry.getValue()))
+        .collect(Collectors.toList());
+  }
+
+  private void assertCustomParametersDontOverwriteDefaultParameters(final Map<String, String> customParameters,
+      final List<Map<String, String>> defaultParametersMaps) {
+    final Set<String> keys = defaultParametersMaps.stream()
+        .map(Map::keySet)
+        .flatMap(Collection::stream)
+        .collect(Collectors.toSet());
+
+    final List<String> duplicateKeys = keys.stream().filter(customParameters::containsKey).toList();
+    if (!duplicateKeys.isEmpty()) {
+      throw new RuntimeException("Cannot overwrite default JDBC parameter " + duplicateKeys);
+    }
+  }
```

Comment:
Nitpick. If a user specifies a parameter with the same value as our internal default, I think we should just let it pass. So this assertion can check the values for each duplicated key, and only throw exception if the value is different. (This should be easier to do when `defaultParametersMaps` is just one map instead of a list of maps, as suggested in the previous comment.)

But I can think of an argument for keeping it as is. So this is a nitpick. Your call.

Reply:
I like it!. Done